lib/server.js

```diff
@@ -280,7 +280,13 @@ function handleRequest(opts, req, resp) {
 
     var remoteAddr = req.headers['x-client-ip'] || req.socket.remoteAddress;
     req.headers['x-client-ip'] = remoteAddr;
-    req.headers['x-forwarded-for'] = remoteAddr;
+    var xff = req.socket.remoteAddress;
+    if (req.headers['x-forwarded-for']) {
+        // Prepend current client to XFF, so that it is passed on in
+        // sub-requests.
+        xff += ', ' + req.headers['x-forwarded-for'];
```

Comment:
Should we append it, not prepend? The format of XFF is `client, proxy1, proxy2` etc. So we need to put the `req.socket.remoteAddr` to the end?